module/zfs/zfs_vnops.c

```diff
@@ -809,8 +812,19 @@ zfs_write(struct inode *ip, uio_t *uio, int ioflag, cred_t *cr)
 		ssize_t tx_bytes;
 		if (abuf == NULL) {
 			tx_bytes = uio->uio_resid;
+			uio->uio_fault_disable = B_TRUE;
 			error = dmu_write_uio_dbuf(sa_get_db(zp->z_sa_hdl),
 			    uio, nbytes, tx);
+			if (error == EFAULT) {
+				dmu_tx_commit(tx);
+				if (uio_prefaultpages(MIN(n, max_blksz), uio)) {
+					break;
+				}
+				continue;
+			} else if (error != 0) {
+				dmu_tx_abort(tx);
+				break;
```

Comment:
You can’t abort an assigned tx, which you might be doing here (can’t get github to expand the context on my phone to confirm).

Reply:
Yes you are right, we have to commit this tx if any error occurred after successfully assigned tx.